tests/test_aio/test_request_processor.py

```diff
@@ -0,0 +1,333 @@
+import pytest
+
+from autoextract.aio.client import RequestProcessor
+from autoextract.aio.errors import QueryError
+
+
+def test_request_processor_without_retries():
+    # Given an initial query with two items
+    initial_query = [
+        {
+            "url": "https://example.org/first",
+            "pageType": "article",
+        },
+        {
+            "url": "https://example.org/second",
+            "pageType": "article",
+        },
+    ]
+
+    # Initialize our request processor with this query
+    request_processor = RequestProcessor(query=initial_query)
+
+    # Since we haven't processed any response yet,
```

Comment:
I like all these comments. Much easy to read the test :-)

Reply:
Thank you!